### PR TITLE
Propagate context for processes in tasks and other fixes

### PR DESCRIPTION
**Pull request by maciej-szlosarczyk**
All changes are documented in their respective commits, please look there for more detailed explanation.

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **maciej-szlosarczyk** after the PR has been reviewed.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: lib/mv_opentelemetry/ecto.ex [hunk 1]
```diff
@@ -73,6 +73,9 @@ defmodule MvOpentelemetry.Ecto do
     span_name = name(config, event, source)
     span_opts = %{start_time: start_time, attributes: all_attributes, kind: :client}
 
+    parent_context = OpentelemetryProcessPropagator.fetch_parent_ctx(1, :"$callers")
```
Comment (maciej-szlosarczyk) [inline lib/mv_opentelemetry/ecto.ex:76]: https://hexdocs.pm/opentelemetry_process_propagator/OpentelemetryProcessPropagator.html#fetch_parent_ctx/2

File: .known_dialyzer_warnings [hunk 2]
```diff
@@ -0,0 +1,2 @@
+lib/mv_opentelemetry/ecto.ex:126: The call 'Elixir.OpenTelemetry.Ctx':attach(_context@1::{'span_ctx',non_neg_integer(),non_neg_integer(),'undefined' | integer(),[{_,_}],'false' | 'true' | 'undefined',boolean(),'false' | 'true' | 'undefined','undefined' | {atom(),_}}) will never return since it differs in the 1st argument from the success typing arguments: (map())
```
Comment (maciej-szlosarczyk) [inline .known_dialyzer_warnings:1]: These are library warnings 🤷

**Review by antonsatin — APPROVED** [review]

**Comment by sauron-droid** [comment]
LGTM label has been added.  <details>Git tree hash: ed1d8a6d022f7f7490092a5e711b60b5a4bcdfba</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/mv-opentelemetry/pull/30#pullrequestreview-925591546" title="Approved">antonsatin</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fmv-opentelemetry).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/mindvalley/mv-opentelemetry/blob/main/OWNERS)~~ [antonsatin]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**